File: src/main/java/com/clinica/atendimento/model/enums/DiaSemana.java
package com.clinica.atendimento.model.enums;

import lombok.Getter;

@Getter
public enum DiaSemana {
    DOMINGO("D"),
    SEGUNDA("2"),
    TERCA("3"),
    QUARTA("4"),
    QUINTA("5"),
    SEXTA("6"),
    SABADO("S");

    private final String codigo;

    DiaSemana(String codigo) {
        this.codigo = codigo;
    }

    public static DiaSemana fromCodigo(String codigo) {
        if (codigo == null) {
            return null;
        }

        for (DiaSemana diaSemana : DiaSemana.values()) {
            if (diaSemana.getCodigo().equals(codigo)) {
                return diaSemana;
            }
        }
        throw new IllegalArgumentException("Código de dia da semana inválido: " + codigo);
    }
}

File: src/main/kotlin/com/clinica/atendimento/model/enums/DiaSemana.kt
package com.clinica.atendimento.model.enums

enum class DiaSemana(val codigo: String) {
    DOMINGO("D"),
    SEGUNDA("2"),
    TERCA("3"),
    QUARTA("4"),
    QUINTA("5"),
    SEXTA("6"),
    SABADO("S");

    companion object {
        @JvmStatic
        fun fromCodigo(codigo: String?): DiaSemana? {
            if (codigo == null) {
                return null
            }

            for (diaSemana in values()) {
                if (diaSemana.codigo == codigo) {
                    return diaSemana
                }
            }
            throw IllegalArgumentException("Código de dia da semana inválido: $codigo")
        }
    }
}